<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Q&A 2 How do you generate synthetic RNA-Seq counts and metadata using R? | rnaseq.knit</title>
  <meta name="description" content="üß¨ RNA-Seq Data Science in R" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Q&A 2 How do you generate synthetic RNA-Seq counts and metadata using R? | rnaseq.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/datainsights/cdi-rnaseq/" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Q&A 2 How do you generate synthetic RNA-Seq counts and metadata using R? | rnaseq.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="prev" href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"/>
<link rel="next" href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://datainsights.github.io/cdi-rnaseq">RNA-SEQ Insights</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a></li>
<li class="part"><span><b>I PREFACE</b></span></li>
<li class="chapter" data-level="" data-path="welcome-to-the-rna-seq-qa-guide.html"><a href="welcome-to-the-rna-seq-qa-guide.html"><i class="fa fa-check"></i>üå± Welcome to the RNA-Seq Q&amp;A Guide</a>
<ul>
<li class="chapter" data-level="" data-path="welcome-to-the-rna-seq-qa-guide.html"><a href="welcome-to-the-rna-seq-qa-guide.html#the-cdi-learning-path"><i class="fa fa-check"></i>üåê The CDI Learning Path</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="what-kind-of-data-will-you-be-using-in-this-guide.html"><a href="what-kind-of-data-will-you-be-using-in-this-guide.html"><i class="fa fa-check"></i>üìÇ What kind of data will you be using in this guide?</a></li>
<li class="part"><span><b>II DATA PREPARATION</b></span></li>
<li class="chapter" data-level="1" data-path="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><i class="fa fa-check"></i><b>1</b> How do you create a project data folder for storing RNA-Seq inputs and outputs?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html#explanation"><i class="fa fa-check"></i><b>1.1</b> Explanation</a></li>
<li class="chapter" data-level="1.2" data-path="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html#r-code"><i class="fa fa-check"></i><b>1.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><i class="fa fa-check"></i><b>2</b> How do you generate synthetic RNA-Seq counts and metadata using R?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#explanation-1"><i class="fa fa-check"></i><b>2.1</b> Explanation</a></li>
<li class="chapter" data-level="2.2" data-path="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#r-code-1"><i class="fa fa-check"></i><b>2.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><i class="fa fa-check"></i><b>3</b> How do you validate RNA-Seq input data before analysis using R?</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#explanation-2"><i class="fa fa-check"></i><b>3.1</b> Explanation</a></li>
<li class="chapter" data-level="3.2" data-path="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#r-code-2"><i class="fa fa-check"></i><b>3.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><i class="fa fa-check"></i><b>4</b> How do you perform differential gene expression analysis using DESeq2 in R?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html#explanation-3"><i class="fa fa-check"></i><b>4.1</b> Explanation</a></li>
<li class="chapter" data-level="4.2" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html#r-code-3"><i class="fa fa-check"></i><b>4.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><i class="fa fa-check"></i><b>5</b> How do you create a volcano plot from DESeq2 results using R?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html#explanation-4"><i class="fa fa-check"></i><b>5.1</b> Explanation</a></li>
<li class="chapter" data-level="5.2" data-path="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html#r-code-4"><i class="fa fa-check"></i><b>5.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><i class="fa fa-check"></i><b>6</b> How do you create an MA plot from DESeq2 results using R?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html#explanation-5"><i class="fa fa-check"></i><b>6.1</b> Explanation</a></li>
<li class="chapter" data-level="6.2" data-path="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html#r-code-5"><i class="fa fa-check"></i><b>6.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><i class="fa fa-check"></i><b>7</b> How do you log-transform RNA-Seq counts for PCA or clustering using R?</a>
<ul>
<li class="chapter" data-level="7.1" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html#explanation-6"><i class="fa fa-check"></i><b>7.1</b> Explanation</a></li>
<li class="chapter" data-level="7.2" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html#r-code-6"><i class="fa fa-check"></i><b>7.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><a href="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><i class="fa fa-check"></i><b>8</b> How do you create a heatmap of top differentially expressed genes using R?</a>
<ul>
<li class="chapter" data-level="8.1" data-path="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><a href="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html#explanation-7"><i class="fa fa-check"></i><b>8.1</b> Explanation</a></li>
<li class="chapter" data-level="8.2" data-path="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><a href="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html#r-code-7"><i class="fa fa-check"></i><b>8.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><a href="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><i class="fa fa-check"></i><b>9</b> How do you visualize RNA-Seq samples using PCA in R?</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><a href="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html#explanation-8"><i class="fa fa-check"></i><b>9.1</b> Explanation</a></li>
<li class="chapter" data-level="9.2" data-path="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><a href="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html#r-code-8"><i class="fa fa-check"></i><b>9.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><i class="fa fa-check"></i><b>10</b> How do you visualize the expression of a single gene across conditions using R?</a>
<ul>
<li class="chapter" data-level="10.1" data-path="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html#explanation-9"><i class="fa fa-check"></i><b>10.1</b> Explanation</a></li>
<li class="chapter" data-level="10.2" data-path="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html#r-code-9"><i class="fa fa-check"></i><b>10.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><i class="fa fa-check"></i><b>11</b> How do you visualize the expression of two or more genes across conditions using R?</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html#explanation-10"><i class="fa fa-check"></i><b>11.1</b> Explanation</a></li>
<li class="chapter" data-level="11.2" data-path="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html#r-code-10"><i class="fa fa-check"></i><b>11.2</b> R Code</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://complexdatainsights.com/product">Explore More Guides</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"><div class="line-block">üß¨ RNA-Seq Data Science in R</div></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Q&A 2</span> How do you generate synthetic RNA-Seq counts and metadata using R?<a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="explanation-1" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Explanation<a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#explanation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this step, we generate <strong>synthetic RNA-Seq data</strong> with known differences between conditions. This allows you to simulate differential expression, save the data into a <code>data/</code> folder, and later analyze it using DESeq2.</p>
<p>We simulate:</p>
<ul>
<li>1000 genes</li>
<li>14 samples (11 Positive, 3 Negative)</li>
<li>Upregulation in the top 30 genes for Negative samples<br />
</li>
<li>Downregulation in the next 30 genes for Negative samples</li>
</ul>
<blockquote>
<p>üîß This setup mimics a real-world study design and ensures that the resulting <strong>volcano and MA plots</strong> clearly show the V-shape pattern of differential expression.</p>
</blockquote>
</div>
<div id="r-code-1" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> R Code<a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#r-code-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-3" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb2-4"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-5" tabindex="-1"></a><span class="co"># üì¶ Create output directory</span></span>
<span id="cb2-6"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-6" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;data&quot;</span>)) <span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-7"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-8" tabindex="-1"></a><span class="co"># üß¨ Simulation settings</span></span>
<span id="cb2-9"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-9" tabindex="-1"></a>n_genes <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb2-10"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-10" tabindex="-1"></a>n_pos <span class="ot">&lt;-</span> <span class="dv">11</span></span>
<span id="cb2-11"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-11" tabindex="-1"></a>n_neg <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb2-12"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-12" tabindex="-1"></a>n_de_up <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb2-13"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-13" tabindex="-1"></a>n_de_down <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb2-14"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-14" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Gene&quot;</span>, <span class="fu">seq_len</span>(n_genes))</span>
<span id="cb2-15"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-15" tabindex="-1"></a></span>
<span id="cb2-16"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-16" tabindex="-1"></a><span class="co"># Simulate Positive group (baseline expression)</span></span>
<span id="cb2-17"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-17" tabindex="-1"></a>counts_pos <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnbinom</span>(n_genes <span class="sc">*</span> n_pos, <span class="at">mu =</span> <span class="dv">100</span>, <span class="at">size =</span> <span class="dv">1</span>), <span class="at">nrow =</span> n_genes)</span>
<span id="cb2-18"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-18" tabindex="-1"></a></span>
<span id="cb2-19"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-19" tabindex="-1"></a><span class="co"># Simulate Negative group</span></span>
<span id="cb2-20"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-20" tabindex="-1"></a>counts_neg <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnbinom</span>(n_genes <span class="sc">*</span> n_neg, <span class="at">mu =</span> <span class="dv">100</span>, <span class="at">size =</span> <span class="dv">1</span>), <span class="at">nrow =</span> n_genes)</span>
<span id="cb2-21"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-21" tabindex="-1"></a></span>
<span id="cb2-22"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-22" tabindex="-1"></a><span class="co"># ‚¨ÜÔ∏è Upregulate top 30 genes in Negative samples</span></span>
<span id="cb2-23"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-23" tabindex="-1"></a>counts_neg[<span class="dv">1</span><span class="sc">:</span>n_de_up, ] <span class="ot">&lt;-</span> counts_neg[<span class="dv">1</span><span class="sc">:</span>n_de_up, ] <span class="sc">+</span> <span class="fu">rnbinom</span>(n_de_up <span class="sc">*</span> n_neg, <span class="at">mu =</span> <span class="dv">400</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb2-24"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-24" tabindex="-1"></a></span>
<span id="cb2-25"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-25" tabindex="-1"></a><span class="co"># ‚¨áÔ∏è Downregulate next 30 genes in Negative samples</span></span>
<span id="cb2-26"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-26" tabindex="-1"></a>counts_neg[(n_de_up <span class="sc">+</span> <span class="dv">1</span>)<span class="sc">:</span>(n_de_up <span class="sc">+</span> n_de_down), ] <span class="ot">&lt;-</span> <span class="fu">rnbinom</span>(n_de_down <span class="sc">*</span> n_neg, <span class="at">mu =</span> <span class="dv">10</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb2-27"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-27" tabindex="-1"></a></span>
<span id="cb2-28"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-28" tabindex="-1"></a><span class="co"># Combine counts</span></span>
<span id="cb2-29"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-29" tabindex="-1"></a>count_matrix <span class="ot">&lt;-</span> <span class="fu">cbind</span>(counts_pos, counts_neg)</span>
<span id="cb2-30"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-30" tabindex="-1"></a><span class="fu">colnames</span>(count_matrix) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Sample&quot;</span>, <span class="fu">seq_len</span>(n_pos <span class="sc">+</span> n_neg))</span>
<span id="cb2-31"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-31" tabindex="-1"></a><span class="fu">rownames</span>(count_matrix) <span class="ot">&lt;-</span> gene_ids</span>
<span id="cb2-32"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-32" tabindex="-1"></a></span>
<span id="cb2-33"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-33" tabindex="-1"></a><span class="co"># üìÑ Metadata</span></span>
<span id="cb2-34"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-34" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2-35"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-35" tabindex="-1"></a>  <span class="at">Sample =</span> <span class="fu">colnames</span>(count_matrix),</span>
<span id="cb2-36"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-36" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Positive&quot;</span>, n_pos), <span class="fu">rep</span>(<span class="st">&quot;Negative&quot;</span>, n_neg))</span>
<span id="cb2-37"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-37" tabindex="-1"></a>)</span>
<span id="cb2-38"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-38" tabindex="-1"></a></span>
<span id="cb2-39"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-39" tabindex="-1"></a><span class="co"># üíæ Save to data/</span></span>
<span id="cb2-40"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-40" tabindex="-1"></a><span class="fu">write_csv</span>(<span class="fu">as.data.frame</span>(count_matrix) <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>), <span class="st">&quot;data/demo_counts.csv&quot;</span>)</span>
<span id="cb2-41"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-41" tabindex="-1"></a><span class="fu">write_csv</span>(metadata, <span class="st">&quot;data/demo_metadata.csv&quot;</span>)</span>
<span id="cb2-42"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-42" tabindex="-1"></a></span>
<span id="cb2-43"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-43" tabindex="-1"></a><span class="co"># üëÅÔ∏è Preview first 5 genes √ó 5 samples</span></span>
<span id="cb2-44"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb2-44" tabindex="-1"></a><span class="fu">as.data.frame</span>(count_matrix)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>      Sample1 Sample2 Sample3 Sample4 Sample5
Gene1     186     175     175      54      54
Gene2      45     101      54      48     287
Gene3      38      12     214      92      86
Gene4     293       0     172      35      85
Gene5     215     377      72      18       3</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb4-1" tabindex="-1"></a><span class="co"># üëÅÔ∏è Preview metadata</span></span>
<span id="cb4-2"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#cb4-2" tabindex="-1"></a><span class="fu">head</span>(metadata, <span class="dv">5</span>)</span></code></pre></div>
<pre><code># A tibble: 5 √ó 2
  Sample  condition
  &lt;chr&gt;   &lt;chr&gt;    
1 Sample1 Positive 
2 Sample2 Positive 
3 Sample3 Positive 
4 Sample4 Positive 
5 Sample5 Positive </code></pre>
<blockquote>
<p>‚úÖ Takeaway: This simulation creates a realistic expression pattern where some genes are clearly upregulated or downregulated in one condition. This structure is ideal for learning DE analysis, producing excellent MA and volcano plots, and testing downstream workflows.</p>
</blockquote>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/datainsights/cdi-rnaseq/edit/main/03-1-rnaseq.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["rnaseq.pdf", "rnaseq.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
