<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Q&A 3 How do you validate RNA-Seq input data before analysis using R? | rnaseq.knit</title>
  <meta name="description" content="üß¨ RNA-Seq Data Science in R" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Q&A 3 How do you validate RNA-Seq input data before analysis using R? | rnaseq.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/datainsights/cdi-rnaseq/" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Q&A 3 How do you validate RNA-Seq input data before analysis using R? | rnaseq.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="prev" href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"/>
<link rel="next" href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://datainsights.github.io/cdi-rnaseq">RNA-SEQ Insights</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a></li>
<li class="part"><span><b>I PREFACE</b></span></li>
<li class="chapter" data-level="" data-path="welcome-to-the-rna-seq-qa-guide.html"><a href="welcome-to-the-rna-seq-qa-guide.html"><i class="fa fa-check"></i>üå± Welcome to the RNA-Seq Q&amp;A Guide</a>
<ul>
<li class="chapter" data-level="" data-path="welcome-to-the-rna-seq-qa-guide.html"><a href="welcome-to-the-rna-seq-qa-guide.html#the-cdi-learning-path"><i class="fa fa-check"></i>üåê The CDI Learning Path</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="what-kind-of-data-will-you-be-using-in-this-guide.html"><a href="what-kind-of-data-will-you-be-using-in-this-guide.html"><i class="fa fa-check"></i>üìÇ What kind of data will you be using in this guide?</a></li>
<li class="part"><span><b>II DATA PREPARATION</b></span></li>
<li class="chapter" data-level="1" data-path="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><i class="fa fa-check"></i><b>1</b> How do you create a project data folder for storing RNA-Seq inputs and outputs?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html#explanation"><i class="fa fa-check"></i><b>1.1</b> Explanation</a></li>
<li class="chapter" data-level="1.2" data-path="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html"><a href="how-do-you-create-a-project-data-folder-for-storing-rna-seq-inputs-and-outputs.html#r-code"><i class="fa fa-check"></i><b>1.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><i class="fa fa-check"></i><b>2</b> How do you generate synthetic RNA-Seq counts and metadata using R?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#explanation-1"><i class="fa fa-check"></i><b>2.1</b> Explanation</a></li>
<li class="chapter" data-level="2.2" data-path="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html"><a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html#r-code-1"><i class="fa fa-check"></i><b>2.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><i class="fa fa-check"></i><b>3</b> How do you validate RNA-Seq input data before analysis using R?</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#explanation-2"><i class="fa fa-check"></i><b>3.1</b> Explanation</a></li>
<li class="chapter" data-level="3.2" data-path="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#r-code-2"><i class="fa fa-check"></i><b>3.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><i class="fa fa-check"></i><b>4</b> How do you perform differential gene expression analysis using DESeq2 in R?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html#explanation-3"><i class="fa fa-check"></i><b>4.1</b> Explanation</a></li>
<li class="chapter" data-level="4.2" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html#r-code-3"><i class="fa fa-check"></i><b>4.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><i class="fa fa-check"></i><b>5</b> How do you create a volcano plot from DESeq2 results using R?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html#explanation-4"><i class="fa fa-check"></i><b>5.1</b> Explanation</a></li>
<li class="chapter" data-level="5.2" data-path="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-a-volcano-plot-from-deseq2-results-using-r.html#r-code-4"><i class="fa fa-check"></i><b>5.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><i class="fa fa-check"></i><b>6</b> How do you create an MA plot from DESeq2 results using R?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html#explanation-5"><i class="fa fa-check"></i><b>6.1</b> Explanation</a></li>
<li class="chapter" data-level="6.2" data-path="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html"><a href="how-do-you-create-an-ma-plot-from-deseq2-results-using-r.html#r-code-5"><i class="fa fa-check"></i><b>6.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><i class="fa fa-check"></i><b>7</b> How do you log-transform RNA-Seq counts for PCA or clustering using R?</a>
<ul>
<li class="chapter" data-level="7.1" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html#explanation-6"><i class="fa fa-check"></i><b>7.1</b> Explanation</a></li>
<li class="chapter" data-level="7.2" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering-using-r.html#r-code-6"><i class="fa fa-check"></i><b>7.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><a href="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><i class="fa fa-check"></i><b>8</b> How do you create a heatmap of top differentially expressed genes using R?</a>
<ul>
<li class="chapter" data-level="8.1" data-path="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><a href="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html#explanation-7"><i class="fa fa-check"></i><b>8.1</b> Explanation</a></li>
<li class="chapter" data-level="8.2" data-path="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html"><a href="how-do-you-create-a-heatmap-of-top-differentially-expressed-genes-using-r.html#r-code-7"><i class="fa fa-check"></i><b>8.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><a href="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><i class="fa fa-check"></i><b>9</b> How do you visualize RNA-Seq samples using PCA in R?</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><a href="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html#explanation-8"><i class="fa fa-check"></i><b>9.1</b> Explanation</a></li>
<li class="chapter" data-level="9.2" data-path="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html"><a href="how-do-you-visualize-rna-seq-samples-using-pca-in-r.html#r-code-8"><i class="fa fa-check"></i><b>9.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><i class="fa fa-check"></i><b>10</b> How do you visualize the expression of a single gene across conditions using R?</a>
<ul>
<li class="chapter" data-level="10.1" data-path="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html#explanation-9"><i class="fa fa-check"></i><b>10.1</b> Explanation</a></li>
<li class="chapter" data-level="10.2" data-path="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-across-conditions-using-r.html#r-code-9"><i class="fa fa-check"></i><b>10.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><i class="fa fa-check"></i><b>11</b> How do you visualize the expression of two or more genes across conditions using R?</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html#explanation-10"><i class="fa fa-check"></i><b>11.1</b> Explanation</a></li>
<li class="chapter" data-level="11.2" data-path="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html"><a href="how-do-you-visualize-the-expression-of-two-or-more-genes-across-conditions-using-r.html#r-code-10"><i class="fa fa-check"></i><b>11.2</b> R Code</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://complexdatainsights.com/product">Explore More Guides</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"><div class="line-block">üß¨ RNA-Seq Data Science in R</div></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-do-you-validate-rna-seq-input-data-before-analysis-using-r" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Q&A 3</span> How do you validate RNA-Seq input data before analysis using R?<a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#how-do-you-validate-rna-seq-input-data-before-analysis-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="explanation-2" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Explanation<a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#explanation-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before proceeding with differential expression analysis, it‚Äôs essential to confirm that your input files are loaded correctly and match in structure. This includes:</p>
<ul>
<li>‚úÖ Ensuring all samples in the metadata are present in the count matrix</li>
<li>‚úÖ Verifying that the matrix is numeric and genes are in rows</li>
<li>‚úÖ Checking for NA or non-finite values</li>
</ul>
<p>This validation step prevents downstream errors and ensures a smooth analysis.</p>
</div>
<div id="r-code-2" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> R Code<a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#r-code-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb6-2"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-2" tabindex="-1"></a></span>
<span id="cb6-3"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-3" tabindex="-1"></a><span class="co"># üîÑ Load the count matrix and metadata</span></span>
<span id="cb6-4"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-4" tabindex="-1"></a>count_df <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/demo_counts.csv&quot;</span>)</span>
<span id="cb6-5"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-5" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/demo_metadata.csv&quot;</span>)</span>
<span id="cb6-6"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-6" tabindex="-1"></a></span>
<span id="cb6-7"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-7" tabindex="-1"></a><span class="co"># üß™ Inspect the data</span></span>
<span id="cb6-8"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb6-8" tabindex="-1"></a><span class="fu">glimpse</span>(count_df)</span></code></pre></div>
<pre><code>Rows: 1,000
Columns: 15
$ gene     &lt;chr&gt; &quot;Gene1&quot;, &quot;Gene2&quot;, &quot;Gene3&quot;, &quot;Gene4&quot;, &quot;Gene5&quot;, &quot;Gene6&quot;, &quot;Gene7&quot;‚Ä¶
$ Sample1  &lt;dbl&gt; 186, 45, 38, 293, 215, 246, 99, 8, 38, 197, 28, 98, 93, 142, ‚Ä¶
$ Sample2  &lt;dbl&gt; 175, 101, 12, 0, 377, 376, 115, 13, 66, 82, 89, 14, 38, 114, ‚Ä¶
$ Sample3  &lt;dbl&gt; 175, 54, 214, 172, 72, 34, 74, 13, 88, 49, 223, 237, 315, 145‚Ä¶
$ Sample4  &lt;dbl&gt; 54, 48, 92, 35, 18, 53, 18, 118, 200, 146, 169, 38, 93, 64, 1‚Ä¶
$ Sample5  &lt;dbl&gt; 54, 287, 86, 85, 3, 81, 369, 96, 1, 12, 101, 41, 135, 102, 82‚Ä¶
$ Sample6  &lt;dbl&gt; 67, 74, 110, 173, 27, 180, 82, 22, 146, 8, 148, 70, 0, 432, 1‚Ä¶
$ Sample7  &lt;dbl&gt; 50, 42, 149, 50, 12, 46, 22, 192, 39, 55, 15, 47, 78, 53, 3, ‚Ä¶
$ Sample8  &lt;dbl&gt; 3, 54, 6, 142, 50, 312, 289, 264, 82, 244, 48, 208, 77, 167, ‚Ä¶
$ Sample9  &lt;dbl&gt; 37, 118, 124, 32, 68, 480, 72, 94, 96, 156, 54, 46, 196, 30, ‚Ä¶
$ Sample10 &lt;dbl&gt; 24, 28, 11, 198, 284, 86, 41, 21, 93, 51, 96, 175, 56, 229, 2‚Ä¶
$ Sample11 &lt;dbl&gt; 38, 42, 121, 138, 0, 46, 43, 66, 266, 14, 18, 96, 56, 9, 166,‚Ä¶
$ Sample12 &lt;dbl&gt; 361, 437, 388, 173, 590, 221, 334, 108, 583, 193, 428, 826, 3‚Ä¶
$ Sample13 &lt;dbl&gt; 238, 314, 324, 984, 1208, 287, 350, 438, 170, 346, 857, 986, ‚Ä¶
$ Sample14 &lt;dbl&gt; 266, 586, 589, 889, 650, 189, 659, 190, 650, 172, 106, 434, 1‚Ä¶</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb8-1" tabindex="-1"></a><span class="fu">glimpse</span>(metadata)</span></code></pre></div>
<pre><code>Rows: 14
Columns: 2
$ Sample    &lt;chr&gt; &quot;Sample1&quot;, &quot;Sample2&quot;, &quot;Sample3&quot;, &quot;Sample4&quot;, &quot;Sample5&quot;, &quot;Samp‚Ä¶
$ condition &lt;chr&gt; &quot;Positive&quot;, &quot;Positive&quot;, &quot;Positive&quot;, &quot;Positive&quot;, &quot;Positive&quot;, ‚Ä¶</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb10-1" tabindex="-1"></a><span class="co"># Check that all sample names in metadata are in counts</span></span>
<span id="cb10-2"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb10-2" tabindex="-1"></a><span class="fu">all</span>(metadata<span class="sc">$</span>Sample <span class="sc">%in%</span> <span class="fu">colnames</span>(count_df))  <span class="co"># Should return TRUE</span></span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-1" tabindex="-1"></a><span class="co"># Set gene names as rownames and confirm dimensions</span></span>
<span id="cb12-2"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-2" tabindex="-1"></a>counts <span class="ot">&lt;-</span> count_df <span class="sc">|&gt;</span></span>
<span id="cb12-3"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-3" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">|&gt;</span></span>
<span id="cb12-4"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-4" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb12-5"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-6" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(metadata<span class="sc">$</span>Sample <span class="sc">%in%</span> <span class="fu">colnames</span>(counts)))</span>
<span id="cb12-7"><a href="how-do-you-validate-rna-seq-input-data-before-analysis-using-r.html#cb12-7" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">ncol</span>(counts) <span class="sc">==</span> <span class="fu">nrow</span>(metadata))</span></code></pre></div>
<blockquote>
<p>‚úÖ <strong>Takeaway:</strong> A quick check of structure, dimension, and sample consistency ensures your data is clean and ready for analysis. This step can help catch common mistakes early, such as misaligned sample names or non-numeric entries.</p>
</blockquote>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-do-you-generate-synthetic-rna-seq-counts-and-metadata-using-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/datainsights/cdi-rnaseq/edit/main/03-1-rnaseq.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["rnaseq.pdf", "rnaseq.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
