<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Q&A 4 How do you perform differential gene expression analysis using DESeq2? | rnaseq.knit</title>
  <meta name="description" content="🧬 RNA-Seq Data Science" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Q&A 4 How do you perform differential gene expression analysis using DESeq2? | rnaseq.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/datainsights/cdi-rnaseq/" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Q&A 4 How do you perform differential gene expression analysis using DESeq2? | rnaseq.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="prev" href="how-do-you-perform-pca-to-explore-sample-similarity.html"/>
<link rel="next" href="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://datainsights.github.io/cdi-rnaseq">RNA-SEQ Insights</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a></li>
<li class="chapter" data-level="" data-path="welcome-to-the-rna-seq-qa-guide.html"><a href="welcome-to-the-rna-seq-qa-guide.html"><i class="fa fa-check"></i>🌱 Welcome to the RNA-Seq Q&amp;A Guide</a></li>
<li class="chapter" data-level="" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i>Getting Started: Preparing RNA-Seq Data for Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="getting-started.html"><a href="getting-started.html#prerequisite"><i class="fa fa-check"></i>Prerequisite</a></li>
<li class="chapter" data-level="" data-path="getting-started.html"><a href="getting-started.html#step-1-run-deseq2-in-r"><i class="fa fa-check"></i>Step 1: Run DESeq2 in R</a></li>
<li class="chapter" data-level="" data-path="getting-started.html"><a href="getting-started.html#output-files"><i class="fa fa-check"></i>Output Files</a></li>
<li class="chapter" data-level="" data-path="getting-started.html"><a href="getting-started.html#quick-glimpse-how-the-demo-data-was-generated"><i class="fa fa-check"></i>Quick Glimpse: How the Demo Data Was Generated</a></li>
</ul></li>
<li class="part"><span><b>I COUNT EXPLORATION</b></span></li>
<li class="chapter" data-level="1" data-path="how-do-you-explore-total-read-counts-per-sample-before-normalization.html"><a href="how-do-you-explore-total-read-counts-per-sample-before-normalization.html"><i class="fa fa-check"></i><b>1</b> How do you explore total read counts per sample before normalization?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-do-you-explore-total-read-counts-per-sample-before-normalization.html"><a href="how-do-you-explore-total-read-counts-per-sample-before-normalization.html#explanation"><i class="fa fa-check"></i><b>1.1</b> Explanation</a></li>
<li class="chapter" data-level="1.2" data-path="how-do-you-explore-total-read-counts-per-sample-before-normalization.html"><a href="how-do-you-explore-total-read-counts-per-sample-before-normalization.html#python-code"><i class="fa fa-check"></i><b>1.2</b> Python Code</a></li>
<li class="chapter" data-level="1.3" data-path="how-do-you-explore-total-read-counts-per-sample-before-normalization.html"><a href="how-do-you-explore-total-read-counts-per-sample-before-normalization.html#r-code"><i class="fa fa-check"></i><b>1.3</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html"><i class="fa fa-check"></i><b>2</b> How do you log-transform RNA-Seq counts for PCA or clustering?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html#explanation-1"><i class="fa fa-check"></i><b>2.1</b> Explanation</a></li>
<li class="chapter" data-level="2.2" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html#python-code-1"><i class="fa fa-check"></i><b>2.2</b> Python Code</a></li>
<li class="chapter" data-level="2.3" data-path="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html"><a href="how-do-you-log-transform-rna-seq-counts-for-pca-or-clustering.html#r-code-1"><i class="fa fa-check"></i><b>2.3</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-do-you-perform-pca-to-explore-sample-similarity.html"><a href="how-do-you-perform-pca-to-explore-sample-similarity.html"><i class="fa fa-check"></i><b>3</b> How do you perform PCA to explore sample similarity?</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-do-you-perform-pca-to-explore-sample-similarity.html"><a href="how-do-you-perform-pca-to-explore-sample-similarity.html#explanation-2"><i class="fa fa-check"></i><b>3.1</b> Explanation</a></li>
<li class="chapter" data-level="3.2" data-path="how-do-you-perform-pca-to-explore-sample-similarity.html"><a href="how-do-you-perform-pca-to-explore-sample-similarity.html#python-code-2"><i class="fa fa-check"></i><b>3.2</b> Python Code</a></li>
<li class="chapter" data-level="3.3" data-path="how-do-you-perform-pca-to-explore-sample-similarity.html"><a href="how-do-you-perform-pca-to-explore-sample-similarity.html#r-code-2"><i class="fa fa-check"></i><b>3.3</b> R Code</a></li>
</ul></li>
<li class="part"><span><b>II DIFFERENTIAL GENE EXPRESSION</b></span></li>
<li class="chapter" data-level="4" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html"><i class="fa fa-check"></i><b>4</b> How do you perform differential gene expression analysis using DESeq2?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#explanation-3"><i class="fa fa-check"></i><b>4.1</b> Explanation</a></li>
<li class="chapter" data-level="4.2" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#python-code-3"><i class="fa fa-check"></i><b>4.2</b> Python Code</a></li>
<li class="chapter" data-level="4.3" data-path="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#r-code-3"><i class="fa fa-check"></i><b>4.3</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html"><a href="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html"><i class="fa fa-check"></i><b>5</b> How do you visualize differentially expressed genes with a volcano plot?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html"><a href="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html#explanation-4"><i class="fa fa-check"></i><b>5.1</b> Explanation</a></li>
<li class="chapter" data-level="5.2" data-path="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html"><a href="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html#python-code-4"><i class="fa fa-check"></i><b>5.2</b> Python Code</a></li>
<li class="chapter" data-level="5.3" data-path="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html"><a href="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html#r-code-4"><i class="fa fa-check"></i><b>5.3</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html"><i class="fa fa-check"></i><b>6</b> How do you visualize the expression of a single gene with a boxplot?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html#explanation-5"><i class="fa fa-check"></i><b>6.1</b> Explanation</a></li>
<li class="chapter" data-level="6.2" data-path="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html#python-code-5"><i class="fa fa-check"></i><b>6.2</b> Python Code</a></li>
<li class="chapter" data-level="6.3" data-path="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html"><a href="how-do-you-visualize-the-expression-of-a-single-gene-with-a-boxplot.html#r-code-5"><i class="fa fa-check"></i><b>6.3</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="how-do-you-summarize-expression-changes-with-an-ma-plot.html"><a href="how-do-you-summarize-expression-changes-with-an-ma-plot.html"><i class="fa fa-check"></i><b>7</b> How do you summarize expression changes with an MA plot?</a>
<ul>
<li class="chapter" data-level="7.1" data-path="how-do-you-summarize-expression-changes-with-an-ma-plot.html"><a href="how-do-you-summarize-expression-changes-with-an-ma-plot.html#explanation-6"><i class="fa fa-check"></i><b>7.1</b> Explanation</a></li>
<li class="chapter" data-level="7.2" data-path="how-do-you-summarize-expression-changes-with-an-ma-plot.html"><a href="how-do-you-summarize-expression-changes-with-an-ma-plot.html#python-code-6"><i class="fa fa-check"></i><b>7.2</b> Python Code</a></li>
<li class="chapter" data-level="7.3" data-path="how-do-you-summarize-expression-changes-with-an-ma-plot.html"><a href="how-do-you-summarize-expression-changes-with-an-ma-plot.html#r-code-6"><i class="fa fa-check"></i><b>7.3</b> R Code</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://complexdatainsights.com/product">Explore More Guides</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"><div class="line-block">🧬 RNA-Seq Data Science</div></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-do-you-perform-differential-gene-expression-analysis-using-deseq2" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Q&A 4</span> How do you perform differential gene expression analysis using DESeq2?<a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#how-do-you-perform-differential-gene-expression-analysis-using-deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="explanation-3" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Explanation<a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#explanation-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once the count matrix and sample metadata are properly loaded and matched, we use <strong>DESeq2</strong> to identify genes that are significantly differentially expressed between experimental conditions.</p>
<p>DESeq2 performs a robust multi-step process:</p>
<ol style="list-style-type: decimal">
<li>📊 <strong>Estimates size factors</strong> to normalize for library depth<br />
</li>
<li>🔁 <strong>Calculates dispersion</strong> for each gene to model biological variability<br />
</li>
<li>⚙️ <strong>Fits a negative binomial generalized linear model (GLM)</strong> to the counts<br />
</li>
<li>🧪 <strong>Tests for significance</strong> using the Wald test (or LRT if specified)<br />
</li>
<li>📉 <strong>Adjusts p-values</strong> using False Discovery Rate (FDR) correction</li>
</ol>
<p>The result is a table of genes with corresponding <strong>log2 fold changes</strong>, <strong>p-values</strong>, and <strong>adjusted p-values</strong>—ready for interpretation and visualization.</p>
</div>
<div id="python-code-3" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Python Code<a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#python-code-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb11-1"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb11-1" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: DESeq2 is R-based; in Python, use rpy2 or export counts and run DE in R</span></span>
<span id="cb11-2"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb11-2" tabindex="-1"></a><span class="co"># Placeholder: Python can handle visualization of DE results after exporting from R</span></span></code></pre></div>
</div>
<div id="r-code-3" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> R Code<a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#r-code-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The code below is store in <code>scripts/res-df.R</code></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb12-2"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb12-3"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-3" tabindex="-1"></a></span>
<span id="cb12-4"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-4" tabindex="-1"></a><span class="co"># 🔄 Set seed for reproducibility</span></span>
<span id="cb12-5"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-5" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb12-6"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-6" tabindex="-1"></a></span>
<span id="cb12-7"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-7" tabindex="-1"></a><span class="co"># 🔹 Load data</span></span>
<span id="cb12-8"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-8" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/demo_counts.csv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-9" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;gene&quot;</span>)</span>
<span id="cb12-10"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-10" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/demo_metadata.csv&quot;</span>)</span>
<span id="cb12-11"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-11" tabindex="-1"></a></span>
<span id="cb12-12"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-12" tabindex="-1"></a><span class="co"># 🧬 Create DESeq2 object</span></span>
<span id="cb12-13"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-13" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb12-14"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-14" tabindex="-1"></a>                              <span class="at">colData =</span> metadata,</span>
<span id="cb12-15"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-15" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb12-16"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-16" tabindex="-1"></a></span>
<span id="cb12-17"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-17" tabindex="-1"></a><span class="co"># ⚙️ Run DE analysis</span></span>
<span id="cb12-18"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-18" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb12-19"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-19" tabindex="-1"></a></span>
<span id="cb12-20"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-20" tabindex="-1"></a><span class="co"># 📋 Extract results</span></span>
<span id="cb12-21"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-21" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb12-22"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-22" tabindex="-1"></a></span>
<span id="cb12-23"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-23" tabindex="-1"></a><span class="co"># 🧼 Clean results</span></span>
<span id="cb12-24"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-24" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res) <span class="sc">%&gt;%</span></span>
<span id="cb12-25"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-25" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-26"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-26" tabindex="-1"></a>  <span class="fu">arrange</span>(padj)</span>
<span id="cb12-27"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-27" tabindex="-1"></a></span>
<span id="cb12-28"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-28" tabindex="-1"></a><span class="co"># 💾 Save for downstream visualization if not already saved</span></span>
<span id="cb12-29"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-29" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;data/deseq2_results.csv&quot;</span>)) {</span>
<span id="cb12-30"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-30" tabindex="-1"></a>  <span class="fu">write_csv</span>(res_df, <span class="st">&quot;data/deseq2_results.csv&quot;</span>)</span>
<span id="cb12-31"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-31" tabindex="-1"></a>}</span>
<span id="cb12-32"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-32" tabindex="-1"></a></span>
<span id="cb12-33"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-33" tabindex="-1"></a><span class="co"># 👁️ Preview top results</span></span>
<span id="cb12-34"><a href="how-do-you-perform-differential-gene-expression-analysis-using-deseq2.html#cb12-34" tabindex="-1"></a><span class="fu">head</span>(res_df, <span class="dv">5</span>)</span></code></pre></div>
<pre><code>        gene  baseMean log2FoldChange    lfcSE       stat    pvalue      padj
1       A1BG  1.116303     -2.9257945 1.810125 -1.6163498 0.1060187 0.6361123
2 SEC24B-AS1  5.020260     -0.8370475 1.221274 -0.6853887 0.4930987 0.9861974
3      A2ML1  2.376188     -2.0927894 2.344610 -0.8925959 0.3720736 0.9861974
4       A1CF 60.556927      0.1956390 1.364227  0.1434065 0.8859692 0.9999971
5      GGACT 18.275000     -0.1999325 1.724350 -0.1159465 0.9076949 0.9999971</code></pre>
<blockquote>
<p>✅ <strong>Takeaway:</strong> DESeq2 provides robust statistical testing for identifying differentially expressed genes. Always inspect and sort results by adjusted p-value (<code>padj</code>) to focus on the most significant findings.</p>
</blockquote>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-do-you-perform-pca-to-explore-sample-similarity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="how-do-you-visualize-differentially-expressed-genes-with-a-volcano-plot.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/datainsights/cdi-rnaseq/edit/main/03-1-rnaseq.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["rnaseq.pdf", "rnaseq.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
